desc/equilibrium/coords.py

```diff
@@ -5,14 +5,143 @@
 import numpy as np
 from termcolor import colored
 
-from desc.backend import jit, jnp, put, while_loop
+from desc.backend import fori_loop, jit, jnp, put, while_loop
 from desc.compute import compute as compute_fun
-from desc.compute import get_transforms
-from desc.grid import ConcentricGrid, LinearGrid, QuadratureGrid
+from desc.compute import data_index, get_transforms
+from desc.grid import ConcentricGrid, Grid, LinearGrid, QuadratureGrid
 from desc.transform import Transform
 from desc.utils import Index
 
 
+def map_coordinates(eq, coords, inbasis, outbasis, tol=1e-6, maxiter=30, rhomin=1e-6):
+    """Given coordinates in inbasis, compute corresponding coordinates in outbasis.
+
+    First solves for the computational coordinates that correspond to inbasis, then
+    evaluates outbasis at those locations.
+
+    NOTE: this function cannot be JIT compiled or differentiated with AD.
+
+    Parameters
+    ----------
+    eq : Equilibrium
+        Equilibrium to use
+    coords : ndarray, shape(k,3)
+        2D array of input coordinates. Each row is a different
```

Comment:
shouldn't this be "each column"? shape(k,3) means there are k rows and 3 columns?

or do you mean coordinate like (R1,phi1,Z1) versus (R2,phi2,Z2) i.e. different points in space?